### PR TITLE
feat(e2e): Add tests for the 'net' set of commands

### PR DESCRIPTION
**Pull request by craciunoiuc**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

To run the tests you will need to be root or run with `sudo` because the commands manipulate network interfaces.

Hopefully this works in the actions, otherwise we can just try to skip them in the action.

### CODE REVIEW

File: test/e2e/cli/net_rm_test.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+			err := cmd.Run()
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(stderr.String()).To(BeEmpty())
+			Expect(stdout.String()).To(MatchRegexp(`^Remove a network\n`))
+		})
+	})
+
+	// Requires root privileges
+	When("invoked with the --help flag", func() {
```
Comment (antoineco) [inline test/e2e/cli/net_rm_test.go]: Duplicate spec

File: test/e2e/cli/net_rm_test.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+			cmd.Args = append(cmd.Args, "--driver", "bridge")
+			cmd.Args = append(cmd.Args, "--network", "172.46.0.1/24")
+			cmd.Args = append(cmd.Args, "test-rm-0")
+
+			err := cmd.Run()
+
+			Expect(err).ToNot(HaveOccurred())
+			Expect(stderr.String()).To(BeEmpty())
+
+			// Reset args
```
Comment (antoineco) [inline test/e2e/cli/net_rm_test.go]: (Applies all tests suites)
This same command is set in the global `BeforeEach`, so instead of overriding it, then reset it, couldn't we simply use another var for the other setup commands and avoid those reset steps?

File: test/e2e/cli/net_up_test.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+		AfterEach(func() {
+			stdout = fcmd.NewIOStream()
+			stderr = fcmd.NewIOStream()
+			cmd = fcmd.NewKraftPrivileged(stdout, stderr, cfg.Path())
+			cmd.Args = append(cmd.Args, "net", "rm", "test-up-1")
+			err := cmd.Run()
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should bring the network up and print it's name", func() {
```
Comment (antoineco) [inline test/e2e/cli/net_up_test.go]: its

File: test/e2e/cli/net_down_test.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+		AfterEach(func() {
+			stdout = fcmd.NewIOStream()
+			stderr = fcmd.NewIOStream()
+			cmd = fcmd.NewKraftPrivileged(stdout, stderr, cfg.Path())
+			cmd.Args = append(cmd.Args, "net", "rm", "test-down-1")
+			err := cmd.Run()
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should bring the network down and print it's name", func() {
```
Comment (antoineco) [inline test/e2e/cli/net_down_test.go]: its

**Review by antoineco — COMMENTED** [review]
General comments:

- All these commands are intertwined inside the test suites. `inspect` depends on `up`, `down` and `rm` for example. It seems like we could have easily covered most of them in a single suite.

- It would have been interesting to test that a unikernel can actually make use of the network created by those commands (e.g. nginx from our catalog).

File: test/e2e/cli/net_rm_test.go (hunk 2 above)
Comment (craciunoiuc) [reply test/e2e/cli/net_rm_test.go]: Makes sense, will do for all

**Comment by craciunoiuc** [comment]
Yes, for the first point, I wanted to use `ifconfig` to do what kraft does but then I realised that it's not really a good, generic way to test. So, better to just throw all commands in there and see where it breaks.

For the second point, it will surely be covered in the `kraft run` tests. Here we can at most add a ping to the bridge to see if it's up

**Comment by antoineco** [comment]
Regarding the first point, what I meant is that you could pretty much keep only the file that's currently called `net_inspect`, and still cover almost all relevant functionalities.
If the goal is to test a lot of combinations of flags I agree that we need a tests suite for every subcommand. However I would focus only on the edge cases in these, to avoid the redundancy of the "happy path".

This is not a request for change btw, just an observation to reduce redundancy in the future. The code is written so let's keep it.

**Comment by craciunoiuc** [comment]
Took a bit more to solve all stuff

I now suffixed every set of commands with the command name. Should make things a bit less confusing + less chance for errors because input/output won't (potentially) spill between commands

**Comment by craciunoiuc** [comment]
aaand tests fail now online

**Comment by craciunoiuc** [comment]
image

Even version fails, strange

**Comment by craciunoiuc** [comment]
and now they seem to pass, wut

**Review by antoineco — APPROVED** [review]
